### PR TITLE
feat(react-scheduler): add Vertical Grouping

### PR DESCRIPTION
**Pull request by AryamnovEugeniy**
Feat #2236

### CODE REVIEW

File: packages/dx-react-scheduler/docs/reference/grouping-state.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ None
 Name | Type | Default | Description
 -----|------|---------|------------
 grouping? | Array&lt;[Grouping](#grouping)&gt; | | Specifies resources to group by.
+groupOrientation? | (viewName: string) => [GroupOrientation](#grouporientation) | Scheduler's grouping orientation: either 'Vertical' or 'Horizontal'.
```
Comment (MaximKudriavtsev) [inline packages/dx-react-scheduler/docs/reference/grouping-state.md]: image

File: packages/dx-react-scheduler/docs/reference/week-view.md [hunk 2]
```diff
@@ -143,6 +153,7 @@ Field | Type | Description
 ------|------|------------
 startDate? | Date | Specifies the cell's start time.
 endDate? | Date | Specifies the cell's end time.
+groupOrientation? | [GroupOrientation](./grouping-state.md#grouporientation) | Scheduler's grouping orientation: either 'Vertical' or 'Horizontal'.
```
Comment (MaximKudriavtsev) [inline packages/dx-react-scheduler/docs/reference/week-view.md]: Very strange see it here)

Comment (AryamnovEugeniy) [reply packages/dx-react-scheduler/docs/reference/week-view.md]: It's necessary to use it along with `endOfGroup` prop to render borders correctly

File: packages/dx-react-scheduler/docs/reference/grouping-panel.md [hunk 3]
```diff
@@ -60,6 +61,19 @@ showHeaderForEveryDate? | boolean | Specifies whether to show group headings for
...
 cellComponent | ComponentType&lt;[GroupingPanel.CellProps](#groupingpanelcellprops)&gt; | A component that renders a cell in a row on the grouping panel.
 
+### GroupingPanel.VerticalLayoutProps
+
+Describes properties passed to a component that renders the grouping panel vertically.
+
+Field | Type | Description
+------|------|------------
+groups | Array&lt;Array&lt;[Group](#group)&gt;&gt; | Specifies the final representation of Scheduler's groups and the order they will be rendered in.
+rowSpan | number | Indicates the number of rows in the Scheduler's timetable.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/reference/grouping-panel.md]:
```suggestion
rowSpan | number | The number of rows in the timetable.
```

File: packages/dx-react-scheduler/docs/reference/grouping-panel.md [hunk 4]
```diff
@@ -60,6 +61,19 @@ showHeaderForEveryDate? | boolean | Specifies whether to show group headings for
...
+viewType | string | Specifies the view the layout is rendered in.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/reference/grouping-panel.md]:
```suggestion
viewType | string | The view in which the layout is rendered.
```

File: packages/dx-react-scheduler/docs/reference/grouping-state.md [hunk 5]
```diff
@@ -32,3 +33,9 @@ Describes grouping options.
 Field | Type | Description
 ------|------|------------
 resourceName | string | The name of the resource by which the appointments are grouped.
+
+### GroupOrientation
+
+Describes grouping orientation.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/reference/grouping-state.md]:
```suggestion
Describes group orientation.
```

File: packages/dx-react-scheduler/docs/reference/grouping-panel.md [hunk 6]
```diff
@@ -76,7 +90,12 @@ Field | Type | Description
 ------|------|------------
 group | [Group](#group) | The group the cell represents.
 colSpan | number | The number of columns the cell spans.
+rowSpan? | number | The number of rows the cell spans.
 left | number | The cell's offset from the left.
+height? | number | The cell's height.
+topOffSet? | number | Specifies the distance between the cell's text and the top edge of the Scheduler.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/reference/grouping-panel.md]:
```suggestion
topOffSet? | number | The distance between the cell's text and the top edge of the Scheduler.
```

File: packages/dx-react-scheduler/docs/reference/grouping-panel.md [hunk 7]
```diff
@@ -76,7 +90,12 @@ Field | Type | Description
...
+textStyle? | object | Specifies the CSS properties to apply to the Cell's text.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/reference/grouping-panel.md]:
```suggestion
textStyle? | object | CSS properties to be applied to the cell's text.
```

File: packages/dx-react-scheduler/docs/reference/month-view.md [hunk 8]
```diff
@@ -33,6 +33,7 @@ layoutComponent | ComponentType&lt;[MonthView.LayoutProps](#monthviewlayoutprops
 dayScaleLayoutComponent | ComponentType&lt;[MonthView.DayScaleLayoutProps](#monthviewdayscalelayoutprops)&gt; | | A component that renders a day scale layout.
 dayScaleCellComponent | ComponentType&lt;[MonthView.DayScaleCellProps](#monthviewdayscalecellprops)&gt; | | A component that renders a day scale cell.
 dayScaleRowComponent | ComponentType&lt;[MonthView.RowProps](#monthviewrowprops)&gt; | | A component that renders a day scale row.
+dayScaleEmptyCellComponent | ComponentType&lt;[MonthView.DayScaleEmptyCellProps](#monthviewdayscaleemptycellprops)&gt; | | A component that renders a day scale empty cell.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/reference/month-view.md]:
```suggestion
dayScaleEmptyCellComponent | ComponentType&lt;[MonthView.DayScaleEmptyCellProps](#monthviewdayscaleemptycellprops)&gt; | | A component that renders an empty cell on the day scale.
```

File: packages/dx-react-scheduler/docs/reference/month-view.md [hunk 9]
```diff
@@ -89,6 +90,14 @@ formatDate | [FormatterFn](scheduler.md#formatterfn) | A function that formats d
 groupingInfo? | Array&lt;[Group](./grouping-panel.md/#group)&gt; | Information about the cell's grouping.
 endOfGroup? | boolean | `true` if this cell is last in its group.
 
+### MonthView.DayScaleEmptyCellProps
+
+Describes properties passed to a component that renders a day scale empty cell.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/reference/month-view.md]:
```suggestion
Describes properties passed to a component that renders an empty cell on the day scale.
```

File: packages/dx-react-scheduler/docs/reference/month-view.md [hunk 10]
```diff
@@ -89,6 +90,14 @@ formatDate | [FormatterFn](scheduler.md#formatterfn) | A function that formats d
...
+
+Field | Type | Description
+------|------|------------
+children? | ReactNode | A React node used to render the row content.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/reference/month-view.md]:
```suggestion
children? | ReactNode | A React node used to render the cell's content.
```

File: packages/dx-react-scheduler/docs/reference/month-view.md [hunk 11]
```diff
@@ -139,6 +149,7 @@ Name | Properties | Description
 MonthView.Layout | [MonthView.LayoutProps](#monthviewlayoutprops) | A component that renders a month view layout.
 MonthView.DayScaleLayout | [MonthView.DayScaleLayoutProps](#monthviewdayscalelayoutprops) | A component that renders a day scale layout.
 MonthView.DayScaleCell | [MonthView.DayScaleCellProps](#monthviewdayscalecellprops) | A component that renders a day scale cell.
+MonthView.DayScaleEmptyCell | [DayView.DayScaleEmptyCellProps](#monthviewdayscaleemptycellprops) | A component that renders a day scale empty cell.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/reference/month-view.md]:
```suggestion
MonthView.DayScaleEmptyCell | [DayView.DayScaleEmptyCellProps](#monthviewdayscaleemptycellprops) | A component that renders an empty cell on the day scale.
```

File: packages/dx-react-scheduler/docs/reference/week-view.md [hunk 12]
```diff
@@ -78,7 +78,12 @@ Describes properties passed to a component that renders a time scale layout.
 Field | Type | Description
 ------|------|------------
 cellsData | Array&lt;Array&lt;[WeekView.CellData](#weekviewcelldata)&gt;&gt; | Specifies the cells meta data.
+groups? | Array&lt;Array&lt;[Group](./grouping-panel.md#group)&gt;&gt; | Specifies the final representation of Scheduler's groups and the order they will be rendered in.
+groupOrientation? | [GroupOrientation](./grouping-state.md#grouporientation) | Scheduler's grouping orientation: either 'Vertical' or 'Horizontal'.
+height? | number | Specifies TimeScale's height in pixels.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/reference/week-view.md]:
```suggestion
height? | number | Specifies the height of the time scale in pixels.
```

File: packages/dx-react-scheduler/docs/reference/week-view.md [hunk 13]
```diff
@@ -78,7 +78,12 @@ Describes properties passed to a component that renders a time scale layout.
...
+showAllDayTitle? | boolean | If `true`, the 'All Day' title will be displayed.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/reference/week-view.md]:
```suggestion
showAllDayTitle? | boolean | Specifies whether to display the "All Day" title.
```

File: packages/dx-react-scheduler/docs/reference/week-view.md [hunk 14]
```diff
@@ -78,7 +78,12 @@ Describes properties passed to a component that renders a time scale layout.
...
 labelComponent | ComponentType&lt;[WeekView.TimeScaleLabelProps](#weekviewtimescalelabelprops)&gt; | A component that renders a time scale label.
+allDayTitleComponent? | ComponentType&lt;[AllDay.TitleCellProps](./all-day-panel.md#alldaypaneltitlecellprops)&gt; | A component that renders a title cell.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/reference/week-view.md]:
```suggestion
allDayTitleComponent? | ComponentType&lt;[AllDay.TitleCellProps](./all-day-panel.md#alldaypaneltitlecellprops)&gt; | A component that renders a cell displaying the "All Day" text.
```

File: packages/dx-react-scheduler/docs/reference/week-view.md [hunk 15]
```diff
@@ -131,8 +138,11 @@ Describes properties passed to a component that renders a time table layout.
 Field | Type | Description
 ------|------|------------
 cellsData | Array&lt;Array&lt;[WeekView.CellData](#weekviewcelldata)&gt;&gt; | Specifies the cells meta data.
+allDayCellsData? | Array&lt;Array&lt;[AllDayPanel.CellData](./all-day-panel.md#alldaypanelcelldata)&gt;&gt; | All day cells meta data.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/reference/week-view.md]:
```suggestion
allDayCellsData? | Array&lt;Array&lt;[AllDayPanel.CellData](./all-day-panel.md#alldaypanelcelldata)&gt;&gt; | Information about cells on the All Day Panel.
```

File: packages/dx-react-scheduler/docs/reference/week-view.md [hunk 16]
```diff
@@ -131,8 +138,11 @@ Describes properties passed to a component that renders a time table layout.
...
 cellComponent | ComponentType&lt;[WeekView.TimeTableCellProps](#weekviewtimetablecellprops)&gt; | A component that renders a time table cell.
 rowComponent | ComponentType&lt;[WeekView.RowProps](#weekviewrowprops)&gt; | A component that renders a time table row.
+allDayCellComponent? | ComponentType&lt;[AllDay.CellProps](./all-day-panel.md#alldaypanelcellprops)&gt; | A component that renders an All Day panel cell.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/reference/week-view.md]:
```suggestion
allDayCellComponent? | ComponentType&lt;[AllDay.CellProps](./all-day-panel.md#alldaypanelcellprops)&gt; | A component that renders a cell on the All Day Panel.
```

File: packages/dx-react-scheduler/docs/reference/week-view.md [hunk 17]
```diff
@@ -131,8 +138,11 @@ Describes properties passed to a component that renders a time table layout.
...
+allDayRowComponent? | ComponentType&lt;[AllDay.RowProps](./all-day-panel.md#alldaypanelrowprops)&gt; | A component that renders an All Day panel row.
```
Comment (RomanTsukanov) [inline packages/dx-react-scheduler/docs/reference/week-view.md]:
```suggestion
allDayRowComponent? | ComponentType&lt;[AllDay.RowProps](./all-day-panel.md#alldaypanelrowprops)&gt; | A component that renders a row on the All Day Panel.
```

File: packages/dx-react-scheduler/docs/reference/grouping-panel.md [hunk 18]
```diff
@@ -60,6 +61,18 @@ showHeaderForEveryDate? | boolean | Specifies whether to show group headings for
 rowComponent | ComponentType&lt;[GroupingPanel.RowProps](#groupingpanelrowprops)&gt; |  A component that renders a row on the grouping panel.
 cellComponent | ComponentType&lt;[GroupingPanel.CellProps](#groupingpanelcellprops)&gt; | A component that renders a cell in a row on the grouping panel.
 
+### GroupingPanel.VerticalLayoutProps
+
+Describes properties passed to a component that renders the grouping panel vertically.
```
Comment (arminal) [inline packages/dx-react-scheduler/docs/reference/grouping-panel.md]:
```suggestion
Describes properties passed to a component, which renders the grouping panel vertically.
```

**Pull request merged**